### PR TITLE
RP-278	Reporting - Make S3 Interactions multi-tenant aware

### PR DESCRIPTION
**Pull request by vgonzalez9**
Using Common TenantPathResolver to set tenant path for S3 and local archives
defaulting "" to keep as much as the same as before.

### CODE REVIEW

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/multitenant/TenantPathService.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * Strategy pattern to find the current tenant storage Path.  The goal being to abstract implementation
+ * variations away such as SecurityContext or TenantContextHolder
+ */
+@Service
+public class TenantPathService {
+    private TenantPathDefaultResolver tenantPathResolver;
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/multitenant/TenantPathService.java]: `final`

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveExamExportService.java [hunk 2]
```diff
@@ -43,15 +49,17 @@ public URI exportExams(final User user, final DistrictSchoolExportReportQuery qu
 
         return location;
     }
-
+    public String getExportBasePath() {
```
Comment (malaffoon) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveExamExportService.java]: scope should be package private not public

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveExamExportService.java [hunk 3]
```diff
@@ -20,15 +22,19 @@
     private static final String FileNameFormat = FileNamePrefix + "-%d.csv";
 
     // store exam exports near the other user reports
-    static final String ExportBasePath = ArchiveBackedReportContentService.BasePath + "EXAM-EXPORT/";
+    // Tenancy handled at the top level
+    private final String ExportBasePath = ArchiveBackedReportContentService.BasePath + "EXAM-EXPORT/";
```
Comment (malaffoon) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveExamExportService.java]: can still be `static`

File: report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveExamExportServiceTest.java [hunk 4]
```diff
@@ -29,45 +30,100 @@
...
-
-    @Before
-    public void setup() {
-        examExportService = new ArchiveExamExportService(repository, archiveService);
+    private final String FileName = "export-exam-1.csv";
+    private final String S3Protocol = "s3://bucket/";
+    private final String DefaultPrefix = "";
+    private final String TenantPrefix = "NV";
+    private String rootPath;
+    private String RelativePath;
```
Comment (malaffoon) [inline report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveExamExportServiceTest.java]: nit: odd variable name casing, should be `relativePath`?

File: report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveExamExportServiceTest.java [hunk 5]
```diff
@@ -29,45 +30,100 @@
...
                 .thenReturn("file://" + RelativePath);
 
         when(repository.export(any(PermissionSource.class), any(DistrictSchoolExportReportQuery.class), anyString()))
                 .thenAnswer((Answer<String>) invocationOnMock -> (String) invocationOnMock.getArguments()[2]);
     }
 
+    private void setupFileArchiveDefaultTest () {
+        setupExportService(DefaultPrefix);
+        setupFileArchiveTest();
+    }
```
Comment (malaffoon) [inline report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveExamExportServiceTest.java]: I admit this is entirely personal opinion but i find these little helpers confusing when trying to read the tests. It seems like inlining the two lines wouldn't be bad and would make it easier to follow the test set up.

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/artifact/AmazonS3ArtifactRepository.java [hunk 6]
```diff
@@ -9,6 +9,10 @@
 import java.io.InputStream;
 import java.text.MessageFormat;
 
+
+/*
+Instance wide, doesn't need to be tenant aware.
+ */
```
Comment (malaffoon) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/artifact/AmazonS3ArtifactRepository.java]: nit: should make this a proper class javadoc (even if it has only this one useful comment in it)

**Comment by vgonzalez9** [comment]
Closing this pull request. Changing how the multi-tenancy is configured, and use updated configuration in a similar fashion to Datasource feature changes.

**Pull request closed without merging**